android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMPBrowserActivity.java

```diff
@@ -40,6 +40,16 @@
   // URL for keyboard search web page presented to user when they add a keyboard in the app.
   private static final String KMP_SEARCH_KEYBOARDS_FORMATSTR = "https://%s/go/android/%s/download-keyboards%s";
   private static final String KMP_SEARCH_KEYBOARDS_LANGUAGES = "/languages/%s";
+
+  // Patterns for determining if a link should be opened in external browser
+  // 1. Host isn't keyman.com (production/staging)
+  // 2. Host is keyman.com but not /keyboards/
+  private static final String INTERNAL_KEYBOARDS_LINK_FORMATSTR = "^http(s)?://(%s|%s)/keyboards(/?.*)?$";
```

Comment:
The regex would have matched a URL ending in `/keyboardsfoo` (which should 404 anyway but is imprecise).

```suggestion
  private static final String INTERNAL_KEYBOARDS_LINK_FORMATSTR = "^http(s)?://(%s|%s)/keyboards(/.*)?$";
```

Reply:
Don't we want to allow for `/keyboards`?

Reply:
Yes, the whole final group is optional so that does allow for that route.

Reply:
Java requires the entire regex to match so in my latest commit I also had to allow for

https://keyman-staging.com/keyboards?embed=android&embed_version=14.0
https://keyman-staging.com/keyboards
https://keyman-staging.com/keyboards/install/foo

And disallow
https://keyman-staging.com/keyboardsfoo